### PR TITLE
Sometimes we might want to ignore auth

### PR DESCRIPTION
**Pull request by jiwalker-usgs**
Let me know what you think, I was running into local issues on CCH and thought this would be easy enough.

### CODE REVIEW

**Comment by jiwalker-usgs** [comment]
@thongsav-usgs if you could take a look when you have a minute.

**Comment by thongsav-usgs** [comment]
:+1: 

I like it. I was going to suggest just disabling the auth filter on the client, but then you don't get roles population like this solution does.

File: auth-client/src/main/java/gov/usgs/cida/auth/client/NullAuthClient.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		roles = new ArrayList<>();
+		try {
+			Context ctx = new InitialContext();
+			String delimitedRoles =  (String) ctx.lookup("java:comp/env/" + AUTH_ROLES_JNDI_NAME);
+			if (delimitedRoles != null) {
+				String[] split = delimitedRoles.split(AUTH_ROLES_DELIMITER);
+				roles.addAll(Arrays.asList(split));
+			}
+		} catch (NamingException ex) {
+			LOG.debug("JNDI name " + AUTH_ROLES_JNDI_NAME + "not set, users will have no roles assigned");
```
Comment (isuftin) [inline auth-client/src/main/java/gov/usgs/cida/auth/client/NullAuthClient.java]: ... "not set" should be ... " not set" :D

File: auth-client/src/main/java/gov/usgs/cida/auth/client/NullAuthClient.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	
+	public static final String AUTH_ROLES_JNDI_NAME = "cida.auth.client.null.roles";
+	public static final String AUTH_ROLES_DELIMITER = ",";
+	
+	private List<String> roles;
+	
+	public NullAuthClient() {
+		roles = new ArrayList<>();
+		try {
+			Context ctx = new InitialContext();
```
Comment (isuftin) [inline auth-client/src/main/java/gov/usgs/cida/auth/client/NullAuthClient.java]: Why not use DynamicReadOnlyProperties here?

Comment (jiwalker-usgs) [inline auth-client/src/main/java/gov/usgs/cida/auth/client/NullAuthClient.java]: Not exactly sure, used the same approach as in AuthClientSingleton

**Pull request merged**